src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package taxi.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.db.Storage;
+import taxi.exception.NoSuchManufacturerException;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers
+                .stream()
+                .filter(man -> man.getId().equals(id))
+                .findAny();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> targetManufacturer = get(manufacturer.getId());
```

Comment:
we don't need to call get here, because we have fully formed object as method input param

Reply:
refactored to use without get()